contracts/Strategy.sol

```diff
@@ -16,32 +13,204 @@ import {
     IERC20,
     Address
 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
 
-// Import interfaces for many popular DeFi projects, or add your own!
-//import "../interfaces/<protocol>/<Interface>.sol";
+import {IKashiPair} from "./sushiswap/kashi-lending/interfaces/IKashiPair.sol";
+import {
+    IBentoBoxV1 as IBentoBox
+} from "./sushiswap/bentobox-sdk/contracts/IBentoBoxV1.sol";
+import {
+    Rebase,
+    RebaseLibrary
+} from "./boringcrypto/boring-solidity/libraries/BoringRebase.sol";
+import {BIERC20} from "./boringcrypto/boring-solidity/interfaces/IERC20.sol";
+import "./sushiswap/sushiswap/interfaces/IMasterChef.sol";
+import "./uniswapv2/interfaces/IUniswapV2Router02.sol";
 
 contract Strategy is BaseStrategy {
     using SafeERC20 for IERC20;
     using Address for address;
     using SafeMath for uint256;
+    using RebaseLibrary for Rebase;
+
+    struct KashiPairInfo {
+        IKashiPair kashiPair;
+        uint256 pid;
+    }
+
+    bool internal isOriginal = true;
+    uint256 internal constant MAX_PAIRS = 5;
+    uint256 internal constant MAX_BPS = 1e4;
+
+    address internal constant DEFAULT_SUSHI_ROUTER =
+        0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F;
+
+    // Kashi constants (apply to MediumRiskPairs)
+    uint256 constant KASHI_MINIMUM_TARGET_UTILIZATION = 7e17; // 70%
+    uint256 constant KASHI_MAXIMUM_TARGET_UTILIZATION = 8e17; // 80%
+    uint256 constant KASHI_UTILIZATION_PRECISION = 1e18;
+
+    IERC20 internal constant weth =
+        IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+    IERC20 internal constant sushi =
+        IERC20(0x6B3595068778DD592e39A122f4f5a5cF09C90fE2);
+    IMasterChef internal constant masterChef =
```

Comment:
if size permits, is not bad that this is public for users to verify

Reply:
👍